src/redux/slices/selectedItemsSlice.ts

```diff
@@ -0,0 +1,39 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Gif } from '../../types';
+
+interface SelectedItemsState {
+  selected: string[];
+  selectedGifs: Gif[];
+}
+
+const initialState: SelectedItemsState = {
+  selected: [],
+  selectedGifs: [],
+};
+
+export const selectedItemsSlice = createSlice({
+  name: 'selectedItems',
+  initialState,
+  reducers: {
+    toggleSelection: (state, action: PayloadAction<Gif>) => {
+      const gif = action.payload;
+      const isSelected = state.selected.includes(gif.id);
+
+      if (isSelected) {
+        state.selected = state.selected.filter((id) => id !== gif.id);
+        state.selectedGifs = state.selectedGifs.filter(
+          (item) => item.id !== gif.id
+        );
+      } else {
+        state.selected.push(gif.id);
+        state.selectedGifs.push(gif);
+      }
+    },
+    clearSelection: (state) => {
+      state.selected = [];
+      state.selectedGifs = [];
+    },
+  },
+});
+
+export const { toggleSelection, clearSelection } = selectedItemsSlice.actions;
```

Comment:
It may be handy to have all selectors for each slice in one place: it's easier to read, use, and maintain them. 
For examle, you can export here
```suggestion
export const { toggleSelection, clearSelection } = selectedItemsSlice.actions;
export const getSelectedItems = (state: RootState) =>
  state.selectedItems.selectedGifs;
export const getIsSelected = (state: RootState, id: string) =>
  state.selectedItems.selected.includes(id);
```


Add export 
`export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;`
in your `store.ts`.

And use it via
`const isSelected = useAppSelector((state) => getIsSelected(state, id));`
in `ResultItemSelect`. And so on